web/src/stream_edit.js

```diff
@@ -430,6 +442,13 @@ export function initialize() {
         },
     );
 
+    $("#streams_overlay_container").on("click", ".stream_settings_header .ind-tab", (e) => {
+        const tab = $(e.target).attr("data-tab-key");
+        const url_fragment = right_section_name_to_url_fragment.get(tab);
+        const sub = get_sub_for_target(e.target);
+        setup_subscriptions_stream_hash(sub, url_fragment);
+    });
```

Comment:
I wanted to put this within the existing tab switcher logic, but it's wrapped up in `stream_edit_toggler` and it was hard to put the relevant code in `callback` there, both for typescript and circular import reasons. So I just put it here instead.

Reply:
Can you be specific about what problem you encountered? In general, I would prefer to see code with circular imports and help you figure out how to fix those than to see code that is perturbed from how you think is most readable trying to work around them. It is generally not hard to good write code without circular imports.

Reply:
I'll try to remember to write out more thoroughly the issues I'm having in the future, especially since most of the time when I do that I fix the problem myself. Which is what I just did :p Pushing with the code now in `stream_edit_toggler`